app/controllers/previousRegistrations/AddPreviousRegistrationController.scala

```diff
@@ -42,67 +45,100 @@ class AddPreviousRegistrationController @Inject()(
                                                    cc: AuthenticatedControllerComponents,
                                                    formProvider: AddPreviousRegistrationFormProvider,
                                                    view: AddPreviousRegistrationView
-                                                 )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with Logging with CompletionChecks {
+                                                 )(implicit ec: ExecutionContext)
+  extends FrontendBaseController
+    with I18nSupport
+    with Logging
+    with CompletionChecks {
 
   private val form = formProvider()
   protected val controllerComponents: MessagesControllerComponents = cc
 
   def onPageLoad(waypoints: Waypoints): Action[AnyContent] = cc.authAndGetData(waypoints.inAmend).async {
-    implicit request =>
+    implicit request: AuthenticatedDataRequest[AnyContent] =>
+
+      val previousRegistrations: Seq[_root_.models.domain.PreviousRegistration] = getPreviousRegistrationsWhenInAmend(waypoints, request)
       getDerivedItems(waypoints, DeriveNumberOfPreviousRegistrations) {
         number =>
           val canAddCountries = number < Country.euCountries.size
-          val previousRegistrations = PreviousRegistrationSummary.row(request.userAnswers, Seq.empty, waypoints, AddPreviousRegistrationPage())
+          val previousRegistrationRows = PreviousRegistrationSummary.row(
+            answers = request.userAnswers,
+            existingPreviousRegistrations = previousRegistrations,
+            waypoints = waypoints,
+            sourcePage = AddPreviousRegistrationPage()
+          )
 
           val failureCall: Seq[PreviousRegistrationDetailsWithOptionalVatNumber] => Future[Result] =
             (incomplete: Seq[PreviousRegistrationDetailsWithOptionalVatNumber]) => {
-              Future.successful(Ok(view(form, waypoints, previousRegistrations, canAddCountries, incomplete)))
+              Future.successful(Ok(view(form, waypoints, previousRegistrationRows, canAddCountries, incomplete)))
```

Comment:
Just renamed to previousRegistrationRows as we have 2 previousRegistrations formats already(ETMP/Ours) and I found it a bit confusing